### PR TITLE
Return processor_id and project_id in get_qcs_objects_for_notebooks

### PR DESCRIPTION
**Pull request by dstrain115**
This adds two fields in the QCSObjectForNotebook

This will allow us some flexibility in google notebooks so as not to specify the processor_id.  This will also be a step in enabling tests for these notebooks.

### CODE REVIEW

File: cirq-google/cirq_google/engine/qcs_notebook_test.py [hunk 1]
```diff
@@ -17,11 +17,13 @@
 
 
 def test_get_device_sampler():
-    result = get_qcs_objects_for_notebook()
-    assert result.device is cg.Sycamore
+    result = get_qcs_objects_for_notebook('not_a_valid_project_name')
```
Comment (95-martin-orion) [inline cirq-google/cirq_google/engine/qcs_notebook_test.py:20]: Can we add tests using the new fields, or would that break some QCS behavior?

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]

File: cirq-google/cirq_google/engine/qcs_notebook_test.py (hunk 1 above)
Comment (viathor) [inline cirq-google/cirq_google/engine/qcs_notebook_test.py:20]: I think it'd be good for the tests to continue to exercise the no-arguments call variant. Perhaps we can do both? (We already exercise passing both arguments on line 28 below.)

**Review by viathor — APPROVED** [review]

**Comment by dstrain115** [comment]
Ok.  This is really tricky to test, since it involves authentication.  If you test it locally and you have application default credentials, you are signed in as well, which further complicates things.  Let me see what I can do.

**Comment by dstrain115** [comment]
Tweaked the tests a bit.  PTAL

File: cirq-google/cirq_google/engine/qcs_notebook.py [hunk 2]
```diff
@@ -63,8 +63,7 @@ def get_qcs_objects_for_notebook(
     # Check for Google Application Default Credentials and run
     # interactive login if the notebook is executed in Colab. In
     # case the notebook is executed in Jupyter notebook or other
-    # IPython runtimes, no interactive login is provided, it is
-    # assumed that the `GOOGLE_APPLICATION_CREDENTIALS` env var is
+
```
Comment (95-martin-orion) [inline cirq-google/cirq_google/engine/qcs_notebook.py:66]: Unintended deletion?

File: cirq-google/cirq_google/engine/qcs_notebook.py [hunk 3]
```diff
@@ -96,17 +100,18 @@ def get_qcs_objects_for_notebook(
             print(f"Available processors: {[p.processor_id for p in processors]}")
             print(f"Using processor: {processor.processor_id}")
         if not project_id:
-            project_id = processor.project_id
+            project_id = getattr(processor, 'project_id', getattr(processor, '_project_name', None))
```
Comment (95-martin-orion) [inline cirq-google/cirq_google/engine/qcs_notebook.py:103]: Should this instead use `is_virtual` to determine the default project_id?

**Review by 95-martin-orion — APPROVED** [review]
Couple of nits but I'm happy with the new tests.

File: cirq-google/cirq_google/engine/qcs_notebook.py (hunk 2 above)
Comment (dstrain115) [reply cirq-google/cirq_google/engine/qcs_notebook.py]: Fixed..

File: cirq-google/cirq_google/engine/qcs_notebook.py (hunk 3 above)
Comment (dstrain115) [reply cirq-google/cirq_google/engine/qcs_notebook.py]: Added this to AbstractLocalProcessor instead.

**Comment by dstrain115** [comment]
Tweaked things a bit so that it only returns virtual or prod, not the PhasedFSim simulator.  Also upped the tests so that it covers all the lines and fixes the pylint stuff.

**Pull request merged**